sched/scheduler/jobRunner.go

```diff
@@ -0,0 +1,267 @@
+
+			// if the attempt to update didn't succeed retry
+			for err != nil && retryableErr(err) && attempts < MAX_RETRY {
+				state, err = jr.saga.EndTask(jr.sagaState, update.taskId, update.data)
+				attempts++
+			}
+
+			// if state transition was successful update internal state
+			if state != nil {
+				jr.sagaState = state
+
+				// state transition was not successful maxed out retries
+				// TODO: for now just panic eventually implement dead letter queue.
+			} else {
+				panic(fmt.Sprintf("Failed to succeesfully log EndTask sagaId: %v, taskId: %v", jr.sagaState.SagaId(), update.taskId))
+			}
+
+		case saga.StartCompTask:
+			attempts := 0
+			state, err := jr.saga.StartCompensatingTask(jr.sagaState, update.taskId, update.data)
+
+			// if the attempt to update didn't succeed retry
+			for err != nil && retryableErr(err) && attempts < MAX_RETRY {
+				state, err = jr.saga.StartCompensatingTask(jr.sagaState, update.taskId, update.data)
+				attempts++
+			}
+
+			// if state transition was successful update internal state
+			if state != nil {
+				jr.sagaState = state
+
+				// state transition was not successful maxed out retries
+				// TODO: for now just panic eventually implement dead letter queue.
+			} else {
+				panic(fmt.Sprintf("Failed to succeesfully log StartCompTask. sagaId: %v, taskId: %v", jr.sagaState.SagaId(), update.taskId))
+			}
+
+		case saga.EndCompTask:
+			attempts := 0
+			state, err := jr.saga.EndCompensatingTask(jr.sagaState, update.taskId, update.data)
+
+			// if the attempt to update didn't succeed retry
+			for err != nil && retryableErr(err) && attempts < MAX_RETRY {
+				state, err = jr.saga.EndCompensatingTask(jr.sagaState, update.taskId, update.data)
+				attempts++
+			}
+
+			// if state transition was successful update internal state
+			if state != nil {
+				jr.sagaState = state
+
+				// state transition was not successful maxed out retries
+				// TODO: for now just panic eventually implement dead letter queue.
+			} else {
+				panic(fmt.Sprintf("Failed to succeesfully log EndCompTask. sagaId: %v, taskId: %v", jr.sagaState.SagaId(), update.taskId))
+			}
+
+		case saga.EndSaga:
+			attempts := 0
+			state, err := jr.saga.EndSaga(jr.sagaState)
+
+			// if the attempt to update didn't succeed retry
+			for err != nil && retryableErr(err) && attempts < MAX_RETRY {
+				state, err = jr.saga.EndSaga(jr.sagaState)
+				attempts++
+			}
+
+			// if state transition was successful update internal state
+			if state != nil {
+				jr.sagaState = state
+				// close the channel no more messages should be coming
+				close(jr.updateChan)
+
+				// state transition was not successful maxed out retries
+				// TODO: for now just panic eventually implement dead letter queue.
+			} else {
+				panic(fmt.Sprintf("Failed to succeesfully log EndSaga. sagaId: %v", jr.sagaState.SagaId()))
+			}
+
+		case saga.AbortSaga:
+			attempts := 0
+			state, err := jr.saga.AbortSaga(jr.sagaState)
```

Comment:
Could saga just take the message to append to the saga log, and so you don't need a switch at all here? You can just send it directly to the saga log?